library/sync/src/main/kotlin/org/cru/godtools/sync/GodToolsSyncService.kt

```diff
@@ -137,9 +137,11 @@ class GodToolsSyncService @VisibleForTesting internal constructor(
 
     suspend fun syncDirtyUserCounters() = executeSync<UserCounterSyncTasks> { syncDirtyCounters() }
     suspend fun syncUserCounters(force: Boolean = false) = executeSync<UserCounterSyncTasks> {
-        val resp = syncCounters(force)
-        coroutineScope.launch { syncDirtyCounters() }
-        resp
+        try {
+            syncCounters(force)
+        } finally {
+            coroutineScope.launch { syncDirtyUserCounters() }
```

Comment:
I switched from using `UserCounterSyncTasks.syncDirtyCounters()` to using `GodToolsSyncService.syncDirtyUserCounters()` here.

They are named very similarly because the `GodToolsSyncService` provides a common wrapper API for all the underlying sync tasks that are internal. By going through the wrapper method we gain usage of the `executeSync` DSL that handles any thrown exceptions for us.

An alternative solution is to call `executeSync` directly from within `coroutineScope.launch { }`